src/ChainSafe.Gaming.WalletConnect/WalletConnectProvider.cs

```diff
@@ -233,6 +242,64 @@ public override async Task<string> Connect()
             }
         }
 
+        private async Task CheckAndSwitchNetwork()
+        {
+            var chainId = GetChainId();
+            if (chainId != $"{ChainModel.EvmNamespace}:{chainConfig.ChainId}")
+            {
+                await PromptNetworkSwitch();
+                UpdateSessionChainId(chainConfig.ChainId);
+            }
+        }
+
+        private void UpdateSessionChainId(string newChainId)
```

Comment:
`newChainId` is not used

Reply:
what does this method do?

Reply:
Well, essentially, from my extensive debugging session, I've figured that the namespaces and accounts, after switching the network, now actually have 2 networks. The first one is the network with whom you've connected with (if you were on cronos, cronos will be the first element in the arrray). With updating the sessions chain id, we're completely removing the chain with which we've connected with originally, leaving us with the correct chain.